playground/TestShop/CatalogService/CatalogService.csproj

```diff
@@ -13,13 +13,13 @@
     <Content Include="Images\**\*" CopyToOutputDirectory="PreserveNewest" CopyToPublishDirectory="PreserveNewest" ExcludeFromSingleFile="true" />
   </ItemGroup>
 
-  <ItemGroup>
-    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" />
-  </ItemGroup>
-
   <ItemGroup>
     <ProjectReference Include="..\CatalogModel\CatalogModel.csproj" />
     <ProjectReference Include="..\TestShop.ServiceDefaults\TestShop.ServiceDefaults.csproj" />
+
+    <!-- Npgsql EF needs to match the same major version as the underlying Npgsql assemblies. -->
+    <PackageReference Include="Npgsql.DependencyInjection" VersionOverride="$(Npgsql8Version)" />
```

Comment:
NIT: should we consider doing the override in a single place (like a directory.build.props so all of these testshop projects can benefit) and that way we reduce the number of places that need to be updated come .net 10?

Reply:
We should only override the PackageRefernce if the app is also referencing Npgsql EF Core. There are only a handful of playground apps use it. We can't do it in a single place without affecting the projects that don't want it.

The real problem here is transitive pinning. Since most projects want Npgsql v9, we use that in our Directory.Packages.props (even when targeting net8 - see #6946). But when an app is targeting net8 -> it needs EF Core 8. And if it uses EF Core 8 -> it needs Ngpsql 8.